components/navigation/Filter.js

```diff
@@ -1,31 +1,36 @@
-import {useState} from 'react';
+import { useState } from 'react';
 import PropTypes from 'prop-types';
 import Select from '../form/Select';
 
-export default function Filter({data, onFilter}) {
-    const [state, setState] = useState('Filter');
-    const categories = data.map(data => {
-        return {
-            value: data.category,
-            text : data.category
-        }
-    });
-    const onFilterApply = (value) => {
-        setState(value);
-        const jobs = data.filter((e) => e.category === value);
-        onFilter(jobs);
+export default function Filter({ data, onFilter }) {
+  const [state, setState] = useState('Filter');
+  let foundCategories = new Set();
+  let categories = [];
+
+  data.forEach(data => {
```

Comment:
It was hard to first spot where was the change as I think you applied some autoformatter.

Please have a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#remove_duplicate_elements_from_the_array. Making sure array of categories has no duplicates can be done in much less complicated way.

Reply:
I tried using that method at first but Set compares objects by reference not equality so the duplicate objects would still exist. I could try to first add all the categories to a set and use map to then generate the objects.

Reply:
I am thinking along the lines of this.

```javascript
 let uniqueCategories = [...new Set(data.map(data => data.category))];
```

then calling map on uniqueCategories 

```javascript
let categories = uniqueCategories.map(category => ({ text: category, value: category }))
```